osutil/mount/mount_linux.go

```diff
@@ -0,0 +1,83 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package mount
+
+import (
+	"syscall"
+)
+
+// MountFlagsToOpts returns the symbolic representation of mount flags.
+func MountFlagsToOpts(flags int) (opts []string, unknown int) {
+	if f := syscall.MS_REMOUNT; flags&f == f {
```

Comment:
Why not a helper though:

```
func flagsToStrings(bitFlags int, mappings []flagMap) (opts []string, unknown int) {
	for _, mapping := range mappings {
		if bitFlags&mapping.flag == mapping.flag {
			bitFlags ^= mapping.flag
			opts = append(opts, mapping.opt)
		}
	}
	return opts, bitFlags
}

var mountFlags = []flagMap{
	{syscall.MS_REMOUNT, "MS_REMOUNT"},
	{syscall.MS_BIND, "MS_BIND"},
	{syscall.MS_REC, "MS_REC"},
	{syscall.MS_RDONLY, "MS_RDONLY"},
	{syscall.MS_SLAVE, "MS_SLAVE"},
}

func MountFlagsToOpts(flags int) (opts []string, unknown int) {
	return flagsToStrings(flags, mountFlags)
}
```

Reply:
Please review this part in #7783 but I just didn't want a loop.

Reply:
I tend to agree with @bboozzoo here (but I left my review in the other PR for clarity)